sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CTESubstitution.scala

```diff
@@ -278,23 +298,17 @@ object CTESubstitution extends Rule[LogicalPlan] {
         if cteRelations.exists(r => plan.conf.resolver(r._1, table)) =>
         throw QueryCompilationErrors.timeTravelUnsupportedError(toSQLId(table))
 
-      case u @ UnresolvedRelation(Seq(table), _, _) =>
```

Comment:
With this pattern match, we only match `UnresolvedRelation` with a single name part (`Seq(table)`). We should keep this behavior in the new method.